File: library/src/test/java/org/coner/crispy_fish/filetype/ecf/EventControlFileAssistantTest.java
package org.coner.crispy_fish.filetype.ecf;

import java.nio.file.*;
import org.assertj.core.api.Assertions;
import org.junit.*;

public class EventControlFileAssistantTest {

    private EventControlFileAssistant eventControlFileAssistant;

    @Before
    public void setup() {
        eventControlFileAssistant = new EventControlFileAssistant();
    }

    @Test
    public void whenIsEcfWithValidPathItShouldReturnTrue() {
        Path path = Paths.get("foo.ecf");

        boolean actual = eventControlFileAssistant.isEventControlFilePath(path);

        Assertions.assertThat(actual).isTrue();
    }

    @Test
    public void whenIsEcfWithInvalidPathItShouldReturnFalse() {
        Path path = Paths.get("foo.wrong");

        boolean actual = eventControlFileAssistant.isEventControlFilePath(path);

        Assertions.assertThat(actual).isFalse();
    }

}

File: library/src/test/kotlin/tech/coner/crispyfish/filetype/ecf/EventControlFileAssistantTest.kt
package tech.coner.crispyfish.filetype.ecf

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import tech.coner.crispyfish.internal.filetype.ecf.EventControlFileAssistant
import java.nio.file.Paths

class EventControlFileAssistantTest {

    @Test
    fun whenIsEcfWithValidPathItShouldReturnTrue() {
        val file = Paths.get("foo.ecf")

        val actual = EventControlFileAssistant.isEventControlFile(file)

        Assertions.assertThat(actual).isTrue()
    }

    @Test
    fun whenIsEcfWithRealPathItShouldReturnTrue() {
        val file = Paths.get("home", "me", "Projects", "Club", "2016", "2016-03-05 points autox 1 danville.ecf")

        val actual = EventControlFileAssistant.isEventControlFile(file)

        Assertions.assertThat(actual).isTrue()
    }

    @Test
    fun whenIsEcfWithInvalidPathItShouldReturnFalse() {
        val file = Paths.get("foo.wrong")

        val actual = EventControlFileAssistant.isEventControlFile(file)

        Assertions.assertThat(actual).isFalse()
    }

}
